TESTS/netsocket/nidd/niddsocket_echotest.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2019, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mbed.h"
+#include "EventFlags.h"
+#include "greentea-client/test_env.h"
+#include "nidd_tests.h"
+#include "unity/unity.h"
+#include "utest.h"
+
+using namespace utest::v1;
+
+namespace {
+const int SIGNAL_SIGIO_RX = 0x1;
+const int SIGNAL_SIGIO_TX = 0x2;
+const int SIGIO_TIMEOUT = 1000; //[ms]
+const int RETRIES = 2;
+
+const double EXPECTED_LOSS_RATIO = 0.0;
+const double TOLERATED_LOSS_RATIO = 0.5;
+
+CellularNonIPSocket *sock;
+EventFlags signals;
+
+const int NIDD_BUFF_SIZE = 100;
+char rx_buffer[NIDD_BUFF_SIZE] = {0};
+char tx_buffer[NIDD_BUFF_SIZE] = {0};
+
+const int PKTS = 11;
+const int pkt_sizes[PKTS] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, NIDD_BUFF_SIZE};
+Timer tc_exec_time;
+int time_allotted;
+}
+
+static void _sigio_handler()
+{
+    signals.set(SIGNAL_SIGIO_RX | SIGNAL_SIGIO_TX);
+}
+
+void NIDDSOCKET_ECHOTEST_NONBLOCK()
```

Comment:
Why do you have some function names all in capitals? This doesn't conform to the coding guidelines ...